File: aspnet-core/src/HRMv2.Application/Configuration/Dto/ConfigurationDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HRMv2.Configuration.Dto
{
    public class ConfigurationDto
    {
        public string EmailHR { get; set; }
        public string GoogleClientId { get; set; }
        public string SecurityCode { get; set; }
        public string EnableNormalLogin { get; set; }
        public SettingDto FinfastService { get; set; }
        public SettingDto ProjectService { get; set; }
        public SettingDto TimesheetService { get; set; }
        public SettingDto IMSService { get; set; }
        public SettingDto TalentService { get; set; }
        public SettingDto VoucherService { get; set; }
        public KomuSettingDto KomuService { get; set; }
        public SettingDto HRMService { get; set; }
        public MezonSettingDto MezonService { get; set; }
        public Oauth2Mezon Oauth2Mezon { get; set; }
        public BotHRMSetting BotHRM { get; set; }
    }

    public class BotHRMSetting
    {
        public string NameBot { get; set; }
        public string ApplicationId { get; set; }
        public string ApplicationToken { get; set; }
        public string UrlSentToken { get; set; }
        public string UrlAuthenticate { get; set; }
    }
    public class Oauth2Mezon
    {
        public string Client_Id { set; get; }
        public string Client_Secret { set; get; }
        public string Grant_Type { get; set; }
        public string Redirect_URI { get; set; }
        public string Url_Oauth2Mezon { get; set; }
        public string Url_UserInfo { get; set; }
    }
    public class KomuSettingDto
    {
        public string BaseAddress { get; set; }
        public string SecurityCode { get; set; }
        public string ChannelIdDevMode { get; set; }
        public string EnableNoticeKomu { get; set; }
    }

    public class MezonSettingDto
    {
        public string DevModeUrl { get; set; }
        public string EnableMezonNotification { get; set; }
    }

    public class SettingDto
    {
        public string BaseAddress { get; set; }
        public string SecurityCode { get; set; }
    }
}

File: angular/src/app/service/model/admin/configuration.dto.ts
import { Oauth2Mezon } from './../../../../shared/AppConsts';


export interface ConfigurationDto {
    imsService: SettingDto,
    talentService: SettingDto,
    projectService: SettingDto,
    timesheetService: SettingDto,
    hrmV2Service: string,
    finfastService: SettingDto,
    komuService:KomuSettingDto,
    hrmService: SettingDto,
    oauth2Mezon: Oauth2Mezon,
    botHRM: BotHRMSetting,
}

export interface BotHRMSetting{
    nameBot: string,
    applicationToken: string,
    applicationId: string,
    urlAuthenticate: string,
    urlSentToken: string,
}
export interface Oauth2Mezon{
    client_Id: string,
    client_Secret: string,
    redirect_URI: string,
    grant_Type: string,
    url_Oauth2Mezon: string,
    url_UserInfo: string,
}
export interface SettingDto {
    baseAddress: string;
    securityCode: string;
}

export interface KomuSettingDto {
    baseAddress: string
    securityCode: string
    channelIdDevMode: string
    enableNoticeKomu: string
}

export interface AutoCreateUpdateDto {
    AutoCreateUpdateTimesheetUser: string
    AutoCreateUpdateIMSUser: string
    AutoCreateUpdateProjectUser: string
    AutoCreateUpdateTalentUser: string
}

export interface LoginConfigDto {
    googleClientId: string
    enableNormalLogin: boolean
    enableLoginMezon: boolean
    enableLoginGoogle: boolean
}
export interface WorkerAutoUpdateAllEmployeeInfoToOtherToolConfigDto {
    runAtHour : string
    enableWorkerAutoUpdateAllEmployeeInfoToOtherToolSetting : boolean
}
export interface EmailSettingDto {
    displayName: string
    defaultAddress: string
    host: string
    port: string
    userName: string
    password: string
    enableSsl: string
    useDefaultCredentials: string
}

export interface NotifyChannelDto {
    notifyPlatform: string,
    itChannel: string,
    payrollChannel: string,
    clanWebhookURL: string,  
}
export interface GetConnectResultDto {
    isConnected: boolean,
    message: string
}
